traversal/iterator/GraphIterator.java

```diff
@@ -122,7 +128,12 @@ private boolean computeFirstBranch(int pos) {
                     else {
                         backTrackCleanUp(pos);
                         answer.remove(toID);
-                        seekStack.addSeeks(edge.from().dependedEdgeOrders());
+                        if (edge.onlyStartsFromRelation()) {
+                            assert fromId.isVariable();
+                            seekStack.addSeeks(iterate(scopedEdgeOrders.get(fromId.asVariable())).toSet());
```

Comment:
change SeekStack.addSeeks() to take in collection and we don't need to double iterate